cmd/geth/main.go

```diff
@@ -378,20 +379,12 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 	// Set contract backend for ethereum service if local node
 	// is serving LES requests.
 	if ctx.GlobalInt(utils.LegacyLightServFlag.Name) > 0 || ctx.GlobalInt(utils.LightServeFlag.Name) > 0 {
-		var ethService *eth.Ethereum
-		if err := stack.Service(&ethService); err != nil {
-			utils.Fatalf("Failed to retrieve ethereum service: %v", err)
-		}
-		ethService.SetContractBackend(ethClient)
+		backend.SetContractBackend(ethClient)
```

Comment:
creates rpc client which talks to the backend's own api

- arrange for rpc server to get created and configured much sooner (when making node `New`, from that point on, it'll be fine to attach to it). Then it'll be possible to set contract backend during configuration of eth/les